charts/k8ssandra-cluster/templates/stargate/stargate.yaml

```diff
@@ -0,0 +1,75 @@
+{{- if .Values.stargate.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-k8ssandra-stargate
+  labels:
+    app: {{ .Release.Name }}-k8ssandra-stargate
+{{ include "k8ssandra-cluster.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.stargate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}-k8ssandra-stargate
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}-k8ssandra-stargate
+    spec:
+      containers:
+        - name: {{ .Release.Name }}-k8ssandra-stargate
+  {{- if eq .Values.stargate.containerImage "auto" }}
+    {{- if eq .Values.stargate.clusterVersion "3.11" }}
+          image: "stargateio/stargate-3_11:v0.0.9"
+    {{- else if eq .Values.stargate.clusterVersion "4.0" }}
+          image: "stargateio/stargate-4_0:v0.0.9"
+    {{- else }}
+          image: "containerImage-auto-did-not-recognize-clusterVersion-{{ .Values.stargate.clusterVersion }}"
+    {{- end }}
+  {{- else }}
+          image: {{ .Values.stargate.containerImage | quote }}
+  {{- end }}
+          ports:
+            - name: graphql
+              containerPort: 8080
+            - name: authorization
+              containerPort: 8081
+            - name: rest
+              containerPort: 8082
+            - name: admin
+              containerPort: 8084
+            - name: metrics
+              containerPort: 8085
+          resources:
+            limits:
+              memory: {{ .Values.stargate.resourceLimits.memory | quote }}
+          env:
+            - name: JAVA_OPTS
+              value: {{ .Values.stargate.javaOpts | quote }}
+            - name: CLUSTER_NAME
+              value: {{ .Values.clusterName | quote }}
+            - name: CLUSTER_VERSION
+              value: {{ .Values.stargate.clusterVersion | quote }}
+            - name: SEED
+              value: "{{ .Values.clusterName }}-seed-service"
```

Comment:
We can use the service name, but it has to be its DNS name which looks like:

```yaml
{{ .Values.clusterName }}-seed-service.{{ .Release.Namespace }}.svc.cluster.local
```

Reply:
The `cluster.local` portion of that is configurable in Kubernetes, so we should make that configurable in the template as well. I'm on it.